### PR TITLE
LibraryElements: Creates usage stats for panels and variables

### LINKED ISSUE #32976: LibraryPanels: Collect usage metrics for Cloud customers
- [ ] How many panels on average are users creating

Maybe use this for as inspiration https://github.com/grafana/grafana/pull/34390

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR adds usage collection for Library Panels and Variables.

**Which issue(s) this PR fixes**:
Fixes #32976
Relates #32912

**Special notes for your reviewer**:
I haven't created one of these before, so not sure I am doing it right. Mimicking @kylebrandt PR here https://github.com/grafana/grafana/pull/34390

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
Looks good!

**Comment by hugohaggmark** [comment]
Hmm, not sure how to solve the failing test here https://drone.grafana.net/grafana/grafana/23081/1/8. How do I add the missing library-element table? Should I move the library-elements migrations to the migrations namespace to avoid circular references? Any ideas @aknuds1 @torkelo?

**Comment by kylebrandt** [comment]
> Hmm, not sure how to solve the failing test here https://drone.grafana.net/grafana/grafana/23081/1/8. How do I add the missing library-element table? Should I move the library-elements migrations to the migrations namespace to avoid circular references? Any ideas @aknuds1 @torkelo?

I think it should be moved to migrations, the main reason is so the migration order is not subject to service loading order. Made an issue for this that includes lib panels https://github.com/grafana/grafana/issues/32912

File: pkg/models/libraryelements.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package models
+
+// LibraryElementKind is used for the kind of library element
+type LibraryElementKind int
+
+const (
+	// PanelElement is used for library elements that are of the Panel kind
+	PanelElement LibraryElementKind = iota + 1
```
Comment (aknuds1) [inline pkg/models/libraryelements.go:8]: Why the convention of starting at `iota + 1`?

**Review by aknuds1 — APPROVED** [review]
LGTM, just left a question.

File: pkg/models/libraryelements.go (hunk 1 above)
Comment (hugohaggmark) [reply pkg/models/libraryelements.go:8]: Because I wanted something else than 0. Is there another way?

**Pull request merged**